pkg/api/v1beta3/dynakube/dynakube_props.go

```diff
@@ -120,12 +120,16 @@ func (dk *DynaKube) TenantUUIDFromConnectionInfoStatus() (string, error) {
 	return "", errors.New("tenant UUID not available")
 }
 
-func (dk *DynaKube) ApiRequestThreshold() time.Duration {
-	if dk.Spec.DynatraceApiRequestThreshold < 0 {
-		dk.Spec.DynatraceApiRequestThreshold = DefaultMinRequestThresholdMinutes
+func (dk *DynaKube) GetDynatraceApiRequestThreshold() int {
+	if dk.Spec.DynatraceApiRequestThreshold == nil || *dk.Spec.DynatraceApiRequestThreshold < 0 {
```

Comment:
If I am not mistaken there happens to be a little issue here:

I applied a dynakube with a threshold of -1, which should fall back then to the default 15. This happens rightly so, however, if you describe the dynakube it states that -1 is stored as a threshold. Lemme show you:

the dynakube describe:

```
│ Name:         dynakube                                                                                                                                                                                           │
│ Namespace:    dynatrace                                                                                                                                                                                          │
│ Labels:       <none>                                                                                                                                                                                             │
│ Annotations:  <none>                                                                                                                                                                                             │
│ API Version:  dynatrace.com/v1beta3                                                                                                                                                                              │
│ Kind:         DynaKube                                                                                                                                                                                           │
│ Metadata:                                                                                                                                                                                                        │
│   Creation Timestamp:  2024-11-05T12:33:39Z                                                                                                                                                                      │
│   Generation:          7                                                                                                                                                                                         │
│   Resource Version:    56374999                                                                                                                                                                                  │
│   UID:                 a8d8132a-135a-41b9-bf0a-72bc85a09d5e                                                                                                                                                      │
│ Spec:                                                                                                                                                                                                            │
│   API URL:                          https://asj34817.dev.dynatracelabs.com/api                                                                                                                                   │
│   Dynatrace API Request Threshold:  -1    
```


the proof that in code it is set to 15:

`│ {"level":"info","ts":"2024-11-06T08:49:05.559Z","logger":"oneagent-connectioninfo","msg":"skipping OneAgent connection info update, last request was made less than 15 minutes ago, 14 minutes remaining until n │`

Reply:
actually, it works as expected, kinda, we should probably deny this input in validation

Reply:
> Also if you do not set a threshold at all, it won't show up in the describe at all (in code again it works)

that is what we want

Reply:
don't we allow a value of `-1` in some places to disable the field, i.e. no request threshold, or do we just say it should be set to `0`?

Reply:
just set it to 0

Reply:
Alright but isn't it somehow misleading if someone for whatever (accidentally for example) reason applies a dynakube with a negative value then describes the dynakube sees the negative value but in code 15 is used? Shouldn't then the threshold show 15 in the describe (maybe with a validation warning, "threshold set to an invalid value, fall to default 15" or something along those lines)?

Reply:
even nicer 👍 I know this probably not even such an important finding but I was still somewhat confused. Thanks for clearing that up

Reply:
>  I will make it an uint

are negative values rejected then or does it error if one is configured?

Reply:
we will see :D

Reply:
now you should get 
```
```